### PR TITLE
podman-remote push

### PR DESCRIPTION
**Pull request by baude**
enable podman-remote push so that users can push images from a
remote client.

change in push API to deal with the need to see output over the
varlink connection.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/varlinkapi/images.go [hunk 1]
```diff
@@ -366,10 +366,59 @@ func (i *LibpodAPI) PushImage(call iopodman.VarlinkCall, name, tag string, tlsVe
...
+	for {
+		line, err := output.ReadString('\n')
+		if err == nil {
+			log = append(log, line)
+			continue
+		} else if err == io.EOF {
+			select {
+			case err := <-c:
+				if err != nil {
+					logrus.Error(err)
```
Comment (jwhonce) [inline pkg/varlinkapi/images.go]: Should this error message embed the image name for better debugging?

File: libpod/adapter/runtime_remote.go [hunk 2]
```diff
@@ -452,3 +454,29 @@ func (r *LocalRuntime) RemoveVolumes(ctx context.Context, c *cliconfig.VolumeRmV
...
+	if err != nil {
+		return err
+	}
+	for {
+		responses, flags, err := reply()
+		if err != nil {
+			return err
+		}
+		for _, line := range responses.Logs {
+			fmt.Print(line)
```
Comment (rhatdan) [inline libpod/adapter/runtime_remote.go]: Do the progress lines on Linux default to stderr?

Comment (baude) [reply libpod/adapter/runtime_remote.go]: they do not ... seems to work nicely

**Comment by mheon** [comment]
LGTM

File: libpod/adapter/runtime_remote.go [hunk 3]
```diff
@@ -452,3 +454,29 @@ func (r *LocalRuntime) RemoveVolumes(ctx context.Context, c *cliconfig.VolumeRmV
...
 	return iopodman.VolumeRemove().Call(r.Conn, rmOpts)
 }
+
+func (r *LocalRuntime) Push(ctx context.Context, srcName, destination, manifestMIMEType, authfile, signaturePolicyPath string, writer io.Writer, forceCompress bool, signingOptions image.SigningOptions, dockerRegistryOptions *image.DockerRegistryOptions, additionalDockerArchiveTags []reference.NamedTagged) error {
+
+	tls := true
+	if dockerRegistryOptions.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue {
+		tls = false
+	}
+	reply, err := iopodman.PushImage().Send(r.Conn, varlink.More, srcName, destination, tls, signaturePolicyPath, "", dockerRegistryOptions.DockerCertPath, forceCompress, manifestMIMEType, signingOptions.RemoveSignatures, signingOptions.SignBy)
```
Comment (TomSweeneyRedHat) [inline libpod/adapter/runtime_remote.go]: Not for this PR, but something to consider.  This is a humongous function signature.  Might be good to create a struct with a number of these within it and pass that instead.  Food for thought.

**Pull request merged**